transport/src/main/java/io/netty/channel/FileRegionHelper.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2013 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel;
+
+import io.netty.util.IllegalReferenceCountException;
+
+import java.io.IOException;
+import java.nio.channels.FileChannel;
+import java.nio.channels.WritableByteChannel;
+
+/**
+ * Helper class for implementing <tt>transferTo</tt> and <tt>slice</tt>.
+ */
+final class FileRegionHelper {
+
+    private FileRegionHelper() {
+    }
+
+    /**
+     * Define methods that used to implement
+     * {@link FileRegionHelper#transferTo(InternalFileRegion, WritableByteChannel, long)} and
+     * {@link FileRegionHelper#slice(InternalFileRegion, DefaultFileRegion, long)}.
+     */
+    interface InternalFileRegion extends FileRegion {
+
+        void incTransfered(long inc);
```

Comment:
@Scottmitch I'm wondering if we need this interface at all.  Since we've added `slice` to `FileRegion`, what if we also included a `readSlice` that performed the position increment?